src/library-aux/scala/Any.scala

```diff
@@ -138,15 +138,107 @@ abstract class Any {
    */
   final def isInstanceOf[T0]: Boolean = sys.error("isInstanceOf")
 
-  /** Cast the receiver object to be of type `T0`.
-   *
-   *  Note that the success of a cast at runtime is modulo Scala's erasure semantics.
-   *  Therefore the expression `1.asInstanceOf[String]` will throw a `ClassCastException` at
-   *  runtime, while the expression `List(1).asInstanceOf[List[String]]` will not.
-   *  In the latter example, because the type argument is erased as part of compilation it is
-   *  not possible to check whether the contents of the list are of the requested type.
+  /** Tell the compiler to treat the receiver object to be of type `T0`.
+   *
+   *  This operation should be considered unsafe and the outcome **undefined**. On a language level it merely 
```

Comment:
There's nothing undefined about the outcome *when the value is actually a value of the target type*. In those cases it is guaranteed to succeed. It's when it's not that the outcome is undefined.

Reply:
on the forum: Really it’s `unsafeWhenNotInstanceOf`. When it’s `instanceof`, it’s perfectly fine.

Reply:
I know, I stated it like that to not give false hopes but I am totally fine with mentioning that. Also, actual use cases would of course make sense to include and maybe someone else can bring some good ones. There are cases like type tagging which fall  in the category of it's not an instanceOf but still are used regularly. Not sure how we should include this.